cumulus_library/studies/core/builder_medication.py

```diff
@@ -24,20 +24,23 @@ def _check_data_in_fields(self, cursor, schema: str):
         }
 
         table = "medicationrequest"
-        base_col = "medicationcodeableconcept"
+        inline_col = "medicationcodeableconcept"
         with base_utils.get_progress_bar(transient=True) as progress:
             task = progress.add_task(
                 "Detecting available medication sources...",
-                total=7,
+                total=3,
             )
 
             # inline medications from FHIR medication
-            data_types["inline"] = sql_utils.is_codeable_concept_populated(
-                schema, table, base_col, cursor
+            data_types["inline"] = sql_utils.is_field_populated(
+                schema=schema,
+                source_table=table,
+                hierarchy=[(inline_col, dict), ("coding", list)],
```

Comment:
I don't know how you feel about this usage of base types being passed around - I went back and forth on whether it was sensible or not.

Reply:
i also like it as is just for ease of extension - like if we ever want to add things for basic fields, they just slot right in.